### PR TITLE
feat(gms/sglang): install current-engine runtime hooks

### PR DESCRIPTION
**Pull request by hutm**
Part 6 of 14 in #14704. Builds on #14710.

## Why

The persistent pool, allocator, and cache adapters must be installed at narrow SGLang construction boundaries without replacing unrelated runtime behavior.

## What changes

- Install the persistent allocator and UnifiedRadixCache adapters during setup.
- Hook the current model-loader and memory-occupation lifecycle.
- Keep the integration opt-in and vanilla behavior unchanged.
- Check expected SGLang API shapes so version drift fails clearly.

This wires the engine integration. Primary/shadow orchestration starts next.

## Validation

Runtime-hook and model-loader tests pass at this prefix; changed Python compiles and formats cleanly.

Next: #14712 adds warm TP1 crash failover.

### CODE REVIEW

File: components/src/dynamo/sglang/tests/test_sglang_failover.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+import time
+import types
+from types import SimpleNamespace
+from unittest.mock import AsyncMock
+
+import pytest
+
+
+def _install_sglang_test_compat_modules() -> None:
+    if "dynamo.sglang.register" not in sys.modules:
```
[the remark this answers is not in the document]
Comment (hutm) [reply components/src/dynamo/sglang/tests/test_sglang_failover.py]: Addressed in 59e01d1840b. The test now imports the real Dynamo SGLang modules only when the test executes and no longer installs collection-time stubs in `sys.modules`, so later tests see the complete `dynamo.sglang.register` API. This also lets marker-only CI collect against an absent or incompatible optional SGLang package.

File: lib/gpu_memory_service/tests/test_sglang_model_loader.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        GMSModelLoader._validate_import_model(Model())
+
+    assert called is False
+
+
+def test_import_model_accepts_model_without_custom_post_load():
+    GMSModelLoader._validate_import_model(torch.nn.Linear(2, 2, device="meta"))
+
+
+def test_import_runtime_state_restores_only_missing_flashinfer_state():
```
[the remark this answers is not in the document]
Comment (hutm) [reply lib/gpu_memory_service/tests/test_sglang_model_loader.py:41]: Addressed in 3f280938dabb. The test now includes an already-restored sibling carrying _flashinfer_trtllm_gemm1_alpha and asserts its preparation callback is not invoked, while the missing sibling is restored.

File: lib/gpu_memory_service/integrations/sglang/patches.py [hunk 3]
```diff
@@ -185,23 +188,344 @@ def patched_alloc_memory_pool(self, *args, **kwargs):
 
         return original_alloc_memory_pool(self, *args, **kwargs)
 
-    ModelRunner.alloc_memory_pool = patched_alloc_memory_pool
+    # New SGLang versions ask the loader for ``preloaded_weights_bytes`` and
+    # apply this correction natively. Keep the wrapper only for older versions;
+    # applying both would double-count imported GMS weights and oversize KV.
+    if not hasattr(ModelRunner, "account_preloaded_weights"):
```
[the remark this answers is not in the document]
Comment (hutm) [reply lib/gpu_memory_service/integrations/sglang/patches.py:194]: No code change for weight accounting: pinned SGLang scheduler.py init_target_memory_pool calls account_preloaded_weights immediately before alloc_memory_pool. Repeating that adjustment in this adapter would double-count. The separate valid test-state concern is addressed in 3f280938dabb.

File: lib/gpu_memory_service/integrations/sglang/__init__.py [hunk 4]
```diff
@@ -68,6 +81,11 @@ def setup_gms(server_args) -> Type["GMSModelLoader"]:
             # predates ServerArgs.override. Remove after that pin reaches 0.5.16+.
             server_args.enable_memory_saver = True
 
+    configure_shared_failover_env()
+
+    # Collective timeouts are tightened only after Scheduler.__init__ has
```
[the remark this answers is not in the document]
Comment (hutm) [reply lib/gpu_memory_service/integrations/sglang/__init__.py]: Fixed in b40b78c655e. The duplicated lifecycle comment was removed; timeout timing remains documented at the hook implementation.

File: lib/gpu_memory_service/integrations/sglang/patches.py [hunk 5]
```diff
@@ -212,26 +536,48 @@ def patch_static_state_for_gms() -> None:
     if _static_state_patched:
         return
 
-    try:
-        from sglang.srt.managers import scheduler_update_weights_mixin as _mixin
+    def _export_noop(model):
+        """NO-OP: GMS preserves buffers via VA-stable unmap/remap."""
```
[the remark this answers is not in the document]
Comment (hutm) [reply lib/gpu_memory_service/integrations/sglang/patches.py]: Addressed in the updated head (9e55045): the nested `_export_noop` docstring is removed; the enclosing function retains the rationale.

File: lib/gpu_memory_service/integrations/sglang/patches.py [hunk 6]
```diff
@@ -212,26 +536,48 @@ def patch_static_state_for_gms() -> None:
...
+        return dict(buffers=[])
 
-        def _export_noop(model):
-            """NO-OP: GMS preserves buffers via VA-stable unmap/remap."""
-            return dict(buffers=[])
+    def _import_noop(model, static_params):
+        """NO-OP: GMS preserves buffers via VA-stable unmap/remap."""
```
[the remark this answers is not in the document]
Comment (hutm) [reply lib/gpu_memory_service/integrations/sglang/patches.py]: Addressed in the updated head (9e55045): the nested `_import_noop` docstring is removed; the enclosing function retains the rationale.
